expertsystem/io/xml/_dump.py

```diff
@@ -87,7 +87,7 @@ def _to_quantum_number_list(
     for type_name, instance in conversion_map.items():
         if instance is None:
             continue
-        if type_name not in ["Charge", "Spin"] and instance == 0:
+        if type_name not in ["Charge", "Spin", "IsoSpin"] and instance == 0:
```

Comment:
Why only Charge and Spins are checked here?

Reply:
In the former situation:
`Charge` and `Spin` were always written out, even if they were were 0 (because it's a required property, also according to the JSON schema)

Now:
We want to write out `IsoSpin` as well _if it is defined_, but not if it is `None`.

It's a bit more technical, because the `Spin` has an `__eq__` method that can compare with `int` and `float`, which messes up the logic in the loop here. But I would say these `_dump` modules a mess anyway, I don't want to spend too much time cleaning that up.

Reply:
Yeah for me it was more about why not the other quantum numbers as well. But I guess you only care about isospin like the PR title suggests.

Reply:
Ah yeah. What this loop does (among others) is not serialise things like baryon number if they are 0, as to keep the YAML/XML small.